README.md

```diff
@@ -261,73 +259,350 @@ The SDK provides a simple interface for all the [media APIs mentioned here](http
-You can also use advanced [searching](https://docs.imagekit.io/api-reference/media-api/list-and-search-files#advanced-search-queries) and sorting using `SearchQuery` and `Sort` function.
 
-```cs
-List<ListAPIResponse> resp = await imagekit
-    .Skip(0)
-    .Limit(3)
-    .Sort("DESC_SIZE")
-    .SearchQuery("tags IN [\"tag1\"]")
-    .ListFilesAsync();
-```
+
 
 **2. Get File Details**
 
 Accepts the file ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-details).
 
 ```cs
-ListAPIResponse resp = await imagekit.GetFileDetailsAsync(fileId);
+Task<ResponseMetaData> resp = await imagekit.GetFileDetailsAsync(fileId);
 ```
 
-**3. Get File Metadata**
+**3. Get File Versions**
 
-Accepts the `fileId` or `URI` and fetches the metadata as per the [API documentation here](https://docs.imagekit.io/api-reference/metadata-api/get-image-metadata-for-uploaded-media-files).
+Accepts the file ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-versions).
+
+```.net
+String fileId = "62a04834c10d49825c6de9e8";
+ResponseMetaData resultFileVersions = ImageKit.getFileVersions(fileId);
 
-```cs
-MetadataResponse resp = await imagekit.GetFileMetadataAsync(<fileId|URI>);
 ```
 
-**4. Update File Details**
+**4. Get File Version details**
 
-Update parameters associated with the File as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/update-file-details).
+Accepts the file ID and version ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-version-details).
+
+```java
+String fileId = "62a04834c10d49825c6de9e8";
+String versionId = "62a04834c10d49825c6de9e8";
+ResponseMetaData resultFileVersionDetails = ImageKit.getFileVersionDetails(fileId, versionId);
 
-```cs
-string[] tags = { "one", "two" };
-ListAPIResponse resp = await imagekit
-    .Tags(tags)
-    .CustomCoordinates("10,10,100,100")
-    .UpdateFileDetailsAsync(fileId);
 ```
 
-**5. Delete File**
+**5. Update File Details**
 
-Delete a file as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/delete-file). The method accepts the file ID of the File that has to be deleted.
+Accepts an object of class `FileUpdateRequest` specifying the parameters to be used to update file details. All parameters specified in the [documentation here] (https://docs.imagekit.io/api-reference/media-api/update-file-details) can be passed via their setter functions to get the results.
 
-```cs
-string resp = await imagekit.DeleteFileAsync(fileId);
+```java
+List<String> tags = new ArrayList<>();
+tags.add("Software");
+tags.add("Developer");
+tags.add("Engineer");
+
+List<String> aiTags = new ArrayList<>();
+aiTags.add("Plant");
+FileUpdateRequest fileUpdateRequest = new FileUpdateRequest("fileId");
+fileUpdateRequest.Tags(tags);
+fileUpdateRequest.AITags=aiTags;
```

Comment:
Please take care of the following in all code snippets throughout this readme and any further docs related to the sample app or its docs. For example:

* Space before and after `=` and overall code formatting.
* Choice of sensible variable names.
* Choice of dummy values. You seem to have picked dummy values from Java SDK; I suggest you pick from Node.js SDK as I find them more sensible. I will change Java's readme soon.

Please follow this instruction for all examples.